provider.go

```diff
@@ -144,28 +144,12 @@ func New(options ...ProviderHandler) (*WasmcloudProvider, error) {
 		sourceLinks: make(map[string]InterfaceLinkDefinition, len(sourceLinks)),
 		targetLinks: make(map[string]InterfaceLinkDefinition, len(targetLinks)),
 	}
-
 	for _, opt := range options {
 		err := opt(provider)
 		if err != nil {
 			return nil, err
 		}
 	}
-
-	for _, link := range sourceLinks {
-		err := provider.putLink(link)
-		if err != nil {
-			logger.Error("failed to put source link", slog.Any("error", err))
-		}
-	}
-
-	for _, link := range targetLinks {
-		err := provider.putLink(link)
-		if err != nil {
-			logger.Error("failed to put target link", slog.Any("error", err))
-		}
-	}
```

Comment:
Since we're removing this part, I think we should assign the provider's source/target links to be the links that we pulled out of the host data map. Thoughts?

Reply:
You mean, update the links in the map after this ?
```
	sourceLinks: make(map[string]InterfaceLinkDefinition, len(sourceLinks)),
        targetLinks: make(map[string]InterfaceLinkDefinition, len(targetLinks)),
```